### ISSUE TITLE
InceptionResNetV2: BatchNormalization layer does not work outside during validation/evaluation

### ISSUE THREAD

**Issue by prokotg** [tensorflow/tensorflow#46414]
TF version : **v2.4.0-0-g582c8d236cb 2.4.0**
Python: 3.6.9/3.8
OS Version: Ubuntu 18.04/Windows 10

`BatchNormalization` layer does not work during evaluation and validation phase in `InceptionResNetV2`. Please note that the same data was used for training, validation and evaluation. Dataset consists of 40 examples of cats/dogs. Dataset is balanced (20cat/20dog)

1. Train history **with** Batch Normalization :

```shell
starting script
starting training
Epoch 1/12
4/4 [==============================] - 21s 2s/step - loss: 1.3195 - accuracy: 0.4600 - val_loss: 0.7235 - val_accuracy: 0.5000
Epoch 2/12
4/4 [==============================] - 3s 715ms/step - loss: 1.8840 - accuracy: 0.5500 - val_loss: 0.7115 - val_accuracy: 0.5000
Epoch 3/12
4/4 [==============================] - 3s 714ms/step - loss: 0.6748 - accuracy: 0.6833 - val_loss: 3.8985 - val_accuracy: 0.5000
Epoch 4/12
4/4 [==============================] - 3s 714ms/step - loss: 0.3799 - accuracy: 0.7900 - val_loss: 7.1380 - val_accuracy: 0.5000
Epoch 5/12
4/4 [==============================] - 3s 718ms/step - loss: 0.2764 - accuracy: 0.9267 - val_loss: 7.7212 - val_accuracy: 0.5000
Epoch 6/12
4/4 [==============================] - 3s 718ms/step - loss: 0.1726 - accuracy: 0.9467 - val_loss: 6.9867 - val_accuracy: 0.5000
Epoch 7/12
4/4 [==============================] - 3s 723ms/step - loss: 0.0975 - accuracy: 1.0000 - val_loss: 5.7449 - val_accuracy: 0.5000
Epoch 8/12
4/4 [==============================] - 3s 724ms/step - loss: 0.0473 - accuracy: 1.0000 - val_loss: 4.4831 - val_accuracy: 0.5000
Epoch 9/12
4/4 [==============================] - 3s 726ms/step - loss: 0.0108 - accuracy: 1.0000 - val_loss: 3.7662 - val_accuracy: 0.5000
Epoch 10/12
4/4 [==============================] - 3s 733ms/step - loss: 0.0060 - accuracy: 1.0000 - val_loss: 3.0623 - val_accuracy: 0.5000
Epoch 11/12
4/4 [==============================] - 3s 732ms/step - loss: 0.0030 - accuracy: 1.0000 - val_loss: 2.4364 - val_accuracy: 0.5000
Epoch 12/12
4/4 [==============================] - 3s 736ms/step - loss: 0.0017 - accuracy: 1.0000 - val_loss: 1.9331 - val_accuracy: 0.5000
starting testing
4/4 [==============================] - 1s 197ms/step - loss: 1.9331 - accuracy: 0.5000
[1.9330692291259766, 0.5]
script complete
```

1. Train history **without** Batch Normalization:
```shell
starting script
starting training
Epoch 1/50
4/4 [==============================] - 14s 1s/step - loss: 0.7497 - accuracy: 0.3833 - val_loss: 0.6936 - val_accuracy: 0.5000
Epoch 2/50
4/4 [==============================] - 2s 674ms/step - loss: 0.7000 - accuracy: 0.4100 - val_loss: 0.6927 - val_accuracy: 0.5000
Epoch 3/50
4/4 [==============================] - 2s 671ms/step - loss: 0.6941 - accuracy: 0.4300 - val_loss: 0.6923 - val_accuracy: 0.5000
Epoch 4/50
4/4 [==============================] - 2s 673ms/step - loss: 0.6930 - accuracy: 0.4633 - val_loss: 0.6913 - val_accuracy: 0.5000
Epoch 5/50
4/4 [==============================] - 2s 677ms/step - loss: 0.6923 - accuracy: 0.4600 - val_loss: 0.6884 - val_accuracy: 0.5000
Epoch 6/50
4/4 [==============================] - 2s 678ms/step - loss: 0.6889 - accuracy: 0.4767 - val_loss: 0.6775 - val_accuracy: 0.5000
Epoch 7/50
4/4 [==============================] - 2s 686ms/step - loss: 0.6793 - accuracy: 0.5033 - val_loss: 0.6452 - val_accuracy: 0.6750
Epoch 8/50
4/4 [==============================] - 2s 679ms/step - loss: 0.6553 - accuracy: 0.7533 - val_loss: 0.6098 - val_accuracy: 0.6750
Epoch 9/50
4/4 [==============================] - 2s 680ms/step - loss: 0.5917 - accuracy: 0.7000 - val_loss: 0.6536 - val_accuracy: 0.6250
Epoch 10/50
4/4 [==============================] - 2s 679ms/step - loss: 0.6050 - accuracy: 0.6333 - val_loss: 0.4657 - val_accuracy: 0.8250
Epoch 11/50
4/4 [==============================] - 2s 683ms/step - loss: 0.5150 - accuracy: 0.7667 - val_loss: 0.4900 - val_accuracy: 0.8250
Epoch 12/50
4/4 [==============================] - 2s 681ms/step - loss: 0.4826 - accuracy: 0.8333 - val_loss: 0.4770 - val_accuracy: 0.7250
Epoch 13/50
4/4 [==============================] - 2s 677ms/step - loss: 0.4328 - accuracy: 0.8267 - val_loss: 0.3850 - val_accuracy: 0.8250
Epoch 14/50
4/4 [==============================] - 2s 683ms/step - loss: 0.3578 - accuracy: 0.8600 - val_loss: 0.2969 - val_accuracy: 0.8500
Epoch 15/50
4/4 [==============================] - 3s 736ms/step - loss: 0.2813 - accuracy: 0.8533 - val_loss: 0.2176 - val_accuracy: 0.9000
Epoch 16/50
4/4 [==============================] - 2s 680ms/step - loss: 0.1951 - accuracy: 0.9100 - val_loss: 0.1658 - val_accuracy: 0.9000
Epoch 17/50
4/4 [==============================] - 2s 677ms/step - loss: 0.2913 - accuracy: 0.8467 - val_loss: 0.2642 - val_accuracy: 0.9000
Epoch 18/50
4/4 [==============================] - 2s 677ms/step - loss: 0.2781 - accuracy: 0.9300 - val_loss: 0.3536 - val_accuracy: 0.8250
Epoch 19/50
4/4 [==============================] - 2s 680ms/step - loss: 0.3337 - accuracy: 0.8200 - val_loss: 0.3058 - val_accuracy: 0.8500
Epoch 20/50
4/4 [==============================] - 2s 669ms/step - loss: 0.3164 - accuracy: 0.8533 - val_loss: 0.2551 - val_accuracy: 0.9000
Epoch 21/50
4/4 [==============================] - 2s 671ms/step - loss: 0.2508 - accuracy: 0.8900 - val_loss: 0.2346 - val_accuracy: 0.9500
Epoch 22/50
4/4 [==============================] - 2s 668ms/step - loss: 0.2010 - accuracy: 0.9800 - val_loss: 0.1793 - val_accuracy: 0.9250
Epoch 23/50
4/4 [==============================] - 2s 671ms/step - loss: 0.1348 - accuracy: 0.9433 - val_loss: 0.1695 - val_accuracy: 0.9250
Epoch 24/50
4/4 [==============================] - 2s 664ms/step - loss: 0.1983 - accuracy: 0.9167 - val_loss: 0.4384 - val_accuracy: 0.8250
Epoch 25/50
4/4 [==============================] - 2s 670ms/step - loss: 0.6678 - accuracy: 0.7767 - val_loss: 0.2072 - val_accuracy: 0.9250
Epoch 26/50
4/4 [==============================] - 2s 667ms/step - loss: 0.3736 - accuracy: 0.7867 - val_loss: 0.3285 - val_accuracy: 0.8750
Epoch 27/50
4/4 [==============================] - 2s 671ms/step - loss: 0.3321 - accuracy: 0.8600 - val_loss: 0.4592 - val_accuracy: 0.7250
Epoch 28/50
4/4 [==============================] - 2s 668ms/step - loss: 0.4189 - accuracy: 0.7733 - val_loss: 0.3381 - val_accuracy: 0.8750
Epoch 29/50
4/4 [==============================] - 2s 671ms/step - loss: 0.2884 - accuracy: 0.9167 - val_loss: 0.2362 - val_accuracy: 0.9250
Epoch 30/50
4/4 [==============================] - 2s 669ms/step - loss: 0.2109 - accuracy: 0.9433 - val_loss: 0.1939 - val_accuracy: 0.9250
Epoch 31/50
4/4 [==============================] - 2s 666ms/step - loss: 0.1309 - accuracy: 0.9633 - val_loss: 0.2058 - val_accuracy: 0.9000
Epoch 32/50
4/4 [==============================] - 2s 673ms/step - loss: 0.2605 - accuracy: 0.8733 - val_loss: 0.2232 - val_accuracy: 0.8750
Epoch 33/50
4/4 [==============================] - 2s 675ms/step - loss: 0.1198 - accuracy: 0.9367 - val_loss: 0.1587 - val_accuracy: 0.9250
Epoch 34/50
4/4 [==============================] - 2s 667ms/step - loss: 0.1421 - accuracy: 0.9567 - val_loss: 0.1156 - val_accuracy: 0.9250
Epoch 35/50
4/4 [==============================] - 2s 669ms/step - loss: 0.0929 - accuracy: 0.9367 - val_loss: 0.0473 - val_accuracy: 1.0000
Epoch 36/50
4/4 [==============================] - 2s 671ms/step - loss: 0.0733 - accuracy: 0.9667 - val_loss: 0.0537 - val_accuracy: 0.9750
Epoch 37/50
4/4 [==============================] - 2s 670ms/step - loss: 0.0925 - accuracy: 0.9833 - val_loss: 0.0829 - val_accuracy: 0.9250
Epoch 38/50
4/4 [==============================] - 2s 667ms/step - loss: 0.1385 - accuracy: 0.9400 - val_loss: 0.4641 - val_accuracy: 0.8500
Epoch 39/50
4/4 [==============================] - 2s 666ms/step - loss: 0.4502 - accuracy: 0.8567 - val_loss: 0.1805 - val_accuracy: 0.9000
Epoch 40/50
4/4 [==============================] - 2s 669ms/step - loss: 0.2034 - accuracy: 0.9200 - val_loss: 0.0498 - val_accuracy: 1.0000
Epoch 41/50
4/4 [==============================] - 2s 667ms/step - loss: 0.0675 - accuracy: 0.9800 - val_loss: 0.1448 - val_accuracy: 0.9250
Epoch 42/50
4/4 [==============================] - 2s 670ms/step - loss: 0.0969 - accuracy: 0.9533 - val_loss: 0.0311 - val_accuracy: 1.0000
Epoch 43/50
4/4 [==============================] - 2s 678ms/step - loss: 0.0396 - accuracy: 1.0000 - val_loss: 0.0047 - val_accuracy: 1.0000
Epoch 44/50
4/4 [==============================] - 2s 678ms/step - loss: 0.0011 - accuracy: 1.0000 - val_loss: 0.0748 - val_accuracy: 0.9750
Epoch 45/50
4/4 [==============================] - 2s 676ms/step - loss: 0.2596 - accuracy: 0.9267 - val_loss: 0.2702 - val_accuracy: 0.9000
Epoch 46/50
4/4 [==============================] - 2s 674ms/step - loss: 0.3273 - accuracy: 0.9167 - val_loss: 0.7219 - val_accuracy: 0.7500
Epoch 47/50
4/4 [==============================] - 2s 672ms/step - loss: 0.5580 - accuracy: 0.7633 - val_loss: 0.1916 - val_accuracy: 0.9500
Epoch 48/50
4/4 [==============================] - 2s 674ms/step - loss: 0.3112 - accuracy: 0.9300 - val_loss: 0.1998 - val_accuracy: 0.9750
Epoch 49/50
4/4 [==============================] - 2s 676ms/step - loss: 0.1679 - accuracy: 0.9533 - val_loss: 0.2122 - val_accuracy: 0.9000
Epoch 50/50
4/4 [==============================] - 2s 675ms/step - loss: 0.1413 - accuracy: 0.9733 - val_loss: 0.1079 - val_accuracy: 0.9500
starting testing
4/4 [==============================] - 1s 184ms/step - loss: 0.1079 - accuracy: 0.9500
[0.10788657516241074, 0.949999988079071]
script complete
```

Link to fully reproduce (with dataset, takes about 3 minutes) :
 https://colab.research.google.com/drive/1QJVZC-2u9aq2RaSCUlmRQeMbZ2QzMHyx?usp=sharing

This was inspired by https://stackoverflow.com/q/65415799/9730862 (I'm not the OP, slightly modified setup)

I know similar issues were raised in the past but taking into account that this behaviour also affects  built-in models  such as `InceptionResNetV2` I'm starting this thread in hope for reveletion. 

Cheers!

**Comment by prokotg** [comment]
It's not limited to TF2.4, just tested it on 2.2

**Comment by Saduf2019** [comment]
@prokotg 
I ran the code on nightly please find the [gist here](https://colab.research.google.com/gist/Saduf2019/d7c95a146344f4df19f6e49d9b0ca639/untitled499.ipynb#scrollTo=5Xg8pFt-Up2I) please let us know if it confirms your issue.

**Comment by prokotg** [comment]
@Saduf2019  Yes, it's the same issue

**Comment by jvishnuvardhan** [comment]
@prokotg Code is too long. I didn't see BatchNormLayer from Keras. Did you implement your own batchnorm layer?

 GitHub is mainly for bugs and performance related issues. This issue looks more like a support issues. We encourage you to post it in Stackoverflow where there is a large community to support this kind of questions. Thanks!
